evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -179,65 +184,105 @@ def _create_naive_pipelines(self, use_features=False):
             feature_selector = []
 
         estimators = self._naive_estimators()
-        parameters = self._pipeline_params if self._pipeline_params else None
         pipelines = [
             make_pipeline(
                 self.X,
                 self.y,
                 estimator,
                 self.problem_type,
                 sampler_name=self.sampler_name,
-                parameters=parameters,
                 extra_components=feature_selector,
+                extra_components_position="after_preprocessing",
             )
             for estimator in estimators
         ]
 
         pipelines = self._create_pipelines_with_params(pipelines, parameters={})
         return pipelines
 
+    def _find_component_names(self, original_name, pipeline):
+        names = []
+        for component in pipeline.component_graph.compute_order:
+            split = component.split(" - ")
+            if len(split) > 1:
+                split = split[1]
+            if original_name in split:
+                names.append(component)
+        return names
+
+    def _create_split_select_parameters(self, pipeline):
+        parameters = {
+            "Categorical Pipeline - Select Columns Transformer": {
+                "columns": self._selected_cat_cols
+            },
+            "Numeric Pipeline - Select Columns Transformer": {
+                "columns": self._selected_cols
+            },
+        }
+        return parameters
+
     def _create_fast_final(self):
         estimators = [
             estimator
             for estimator in get_estimators(self.problem_type)
             if estimator not in self._naive_estimators()
         ]
-        parameters = self._pipeline_params if self._pipeline_params else {}
-        parameters.update(
-            {"Select Columns Transformer": {"columns": self._selected_cols}}
-        )
+
         pipelines = [
-            make_pipeline(
-                self.X,
-                self.y,
+            self._make_split_pipeline(
                 estimator,
-                self.problem_type,
-                sampler_name=self.sampler_name,
-                parameters=parameters,
-                extra_components=[SelectColumns],
             )
             for estimator in estimators
         ]
 
-        pipelines = self._create_pipelines_with_params(
-            pipelines, {"Select Columns Transformer": {"columns": self._selected_cols}}
-        )
+        if self._split:
+            # rename all pipeline params with correct component names
```

Comment:
I think this is worth it's own method, especially if we'll eventually have to do the same for hyperparams.